hw09-aop/src/main/java/ru/otus/merets/watcher/Watcher.java

```diff
@@ -0,0 +1,30 @@
+package ru.otus.merets.watcher;
+
+import ru.otus.merets.blackbox.BoxInterface;
+import ru.otus.merets.exception.CreateWatcherException;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Proxy;
+
+
+public class Watcher<T> {
+    public <T> T createMyClass(Class<? super T> childClass) throws CreateWatcherException {
+        Constructor<? super T> contructor = null;
+        try {
+            contructor = childClass.getConstructor();
+        } catch (NoSuchMethodException e) {
+            throw new CreateWatcherException("Cannot get constructor");
+        }
+        Object testObject = null;
+        try {
+            testObject = childClass.cast(contructor.newInstance());
+        } catch (Exception e) {
+            throw new CreateWatcherException("Cannot cast type");
```

Comment:
```suggestion
            throw new CreateWatcherException("Cannot cast type", e);
```

Правильно, что сделали свое исключение.
В него лучше передавать еще оригинальное исключение, чтобы можно было добраться до причины ошибки.

В `CreateWatcherException` еще определить конструктор:
```java
public CreateWatcherException(String message, Throwable cause) {
    super(message, cause);
}
```